libafl/src/monitors/mod.rs

```diff
@@ -143,25 +143,31 @@ impl ClientStats {
 
         self.last_execs_per_sec =
             self.last_execs_per_sec * (1.0 - 1.0 / 16.0) + cur_avg * (1.0 / 16.0);
-        self.last_execs_per_sec as u64
+        self.last_execs_per_sec
     }
 
     /// Get the calculated executions per second for this client
-    #[allow(clippy::cast_sign_loss, clippy::cast_precision_loss)]
+    #[allow(clippy::cast_precision_loss, clippy::cast_sign_loss)]
     #[cfg(not(feature = "afl_exec_sec"))]
-    pub fn execs_per_sec(&mut self, cur_time: Duration) -> u64 {
+    pub fn execs_per_sec_f64(&mut self, cur_time: Duration) -> f64 {
         if self.executions == 0 {
-            return 0;
+            return 0.0;
         }
 
         let elapsed = cur_time
             .checked_sub(self.last_window_time)
             .map_or(0.0, |d| d.as_secs_f64());
         if elapsed as u64 == 0 {
-            return 0;
+            return 0.0;
         }
 
-        ((self.executions as f64) / elapsed) as u64
+        (self.executions as f64) / elapsed
+    }
+
+    #[allow(clippy::cast_sign_loss)]
+    /// Get the calculated executions per second for this client
+    pub fn execs_per_sec(&mut self, cur_time: Duration) -> u64 {
```

Comment:
I think we should just change the api to f64 here, callers can still cast it if they want

Reply:
I wanted not to change any interfaces so that other implementors need no changes. I do not know the contribution guideline, so I've tried taking a safe approach.
I'll rework it per your suggestion.

Reply:
Thanks! Good point, however the next release will be going from 0.8 to 0.9, so it's okay to break the API here. :)

Reply:
Oh, great!
I'm looking around for the uses of the `execs_per_sec()` function, and a related question popped up.

How would you go about the unsigned->float conversion in the case of exporting to TOML and JSON? It seems more API-breaking than code breaking change if the machine-readable output would be changed -- which tools parse the output from `monitors/disk.rs`?

Reply:
Afaik JSON doesn't know integers anyway, so it should be fine. It might make sense to limit the value to two decimals, or something like that?

Reply:
It's alright, JSON does not distuingish numbers, but JSON *parsers* might have trouble deserializing to integers. E.g. python or other language parsers expect schemas to distinguish integers and floats. I'm good at overthinking though :'D

Push theses changes.

Reply:
Oh, I have not added the limiting the decimal places

Reply:
Eventually, we probably want a proper print function, below 10: show decimals, above 1000, add k, above 1m, add m, etc.

Reply:
(for human readable output)

Reply:
Can you explain it a bit more in detail?
Also, there are many printing functions (as you can see it in this commit) 
Maybe a human-readable version of execs_per_sec function (which returns a formatted string) would not hurt then.

Reply:
Yes exactly, that is probably what we want.
Not for json of course 🙃